spec/models/manageiq/providers/openstack/network_manager/cloud_subnet_refresh_spec.rb

```diff
@@ -0,0 +1,75 @@
+describe ManageIQ::Providers::Openstack::NetworkManager::CloudSubnet do
+  describe "refresh" do
+    before do
+      parent_ems = FactoryGirl.create(:ems_openstack_with_authentication)
+      @ems = parent_ems.network_manager
+      EvmSpecHelper.local_miq_server(:zone => Zone.seed)
+    end
+
+    it "removes subnet link to router if the interface is deleted" do
+      setup_mocked_collector
+
+      EmsRefresh.refresh(@ems)
```

Comment:
we will also need targeted refresh spec, checking refresh of the subnet is not nullifying the relation